modules/auxiliary/gather/ldap_query.rb

```diff
@@ -263,6 +269,36 @@ def output_data_csv(entries)
     generate_rex_tables(entries, 'csv')
   end
 
+  def find_root_dn(ldap)
+    filter = '(objectClass=*)'
+    attributes = ['instanceType']
+    @root_dn = @base_dn
+    loop do
```

Comment:
This loop makes me uncomfortable with there not being a clear predicate here that this will ever terminate 🤔

Reply:
The idea here was that we progressively change `@root_dn` and use that as the termination point for the loop, aka if it ever gets to two parts separated by a comma, or does not contain a comma at all, then we terminate the loop.

Reply:
Removed this and loop and the constants with the revised update.